Framework/Geometry/src/Objects/CSGObject.cpp

```diff
@@ -1077,26 +1077,47 @@ int CSGObject::procString(const std::string &lineStr) {
  * @return Number of segments added
  */
 int CSGObject::interceptSurface(Geometry::Track &track) const {
-  int originalCount = track.count(); // Number of intersections original track
-  // Loop over all the surfaces.
+  // Number of intersections original track
+  int originalCount = track.count();
+
+  // Loop over all the surfaces to get the intercepts, i.e. populating
+  // points into LI
   LineIntersectVisit LI(track.startPoint(), track.direction());
   for (auto &surface : m_surList) {
     surface->acceptVisitor(LI);
   }
+
+  // Call the pruner so that we don't have to worry about the duplicates and
+  // the order
+  LI.pruneTrack();
```

Comment:
A clearer name would be `LI.sortAndRemoveDuplicates()`. It is overly wordy, but there is no uncertainty in what the method is doing.

Reply:
got it, the func name is updated with suggested names.